classes/privacy/provider.php

```diff
@@ -0,0 +1,47 @@
+<?php
+// This file is part of the ocis repository for Moodle - http://moodle.org/
+//
+// The ocis repository for Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The ocis repository for Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with the ocis repository for Moodle.  If not, see <http://www.gnu.org/licenses/>.
+//
```

Comment:
This has to be exactly the comment for all moodle files, not adapted to the ocis repository:

```
// This file is part of Moodle - https://moodle.org/
//
// Moodle is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// Moodle is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with Moodle. If not, see <https://www.gnu.org/licenses/>.
```
This should also be fixed for all other php files in the repository.

Unfortunately, I cannot find the part of the moodle dev docs where this requirement is being described, but I've never seen a moodle plugin have something else than exactly this declaration without any changes no matter which plugin that is :)

Reply:
This file will not be part of Moodle, so how can we claim that?
It's part of an additional plugin, that will (most likely) not be in the Moodle Git repo.
I would even suggest to change `http://moodle.org/` to the URL of this repo or similar

Reply:
I asked in the moodle dev chat and it seems to be a requirement for all moodle core files. Important is that all of the content must be GPL, so there has to be a GPL license like you did. It's probably fine like you did it, I just never saw a plugin not just copying this part from moodle core. To be precise: The plugin is useless without a moodle around it, so it effectively is part of moodle, at least that's how I understand it.